src/Display/Slider/Slider.test.tsx

```diff
@@ -0,0 +1,311 @@
+      const { asFragment } = render(
+        <SliderComponent arrowWhite={arrowWhite} />
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  [true, false].forEach(arrowWhite => matchSnapshot(arrowWhite));
+});
+
+describe('<Slider/> prop removeDots', () => {
+  const matchSnapshot = (removeDots: boolean) => {
+    it(`should match snapshot when removeDots ${removeDots} is passed`, async () => {
+      const { asFragment } = render(
+        <SliderComponent removeDots={removeDots} />
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  [true, false].forEach(removeDots => matchSnapshot(removeDots));
+});
+
+describe('<Slider/> prop autoplay', () => {
+  it(`should go to next slide after 6000ms when autoplay is truthy`, () => {
+    jest.useFakeTimers();
+    const { asFragment } = render(<SliderComponent autoplay={true} />);
+
+    // validate initial active slide
+    expect(
+      document.querySelectorAll('li')[0].classList.contains('active')
+    ).toBeTruthy();
+    expect(asFragment()).toMatchSnapshot();
+
+    // validate autoplay
+    jest.advanceTimersByTime(6000);
+    expect(
+      document.querySelectorAll('li')[1].classList.contains('active')
+    ).toBeTruthy();
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it(`should go to the first slide after 6000ms when autoplay is truthy and the current slide is the last slide`, () => {
+    jest.useFakeTimers();
+    const { asFragment } = render(
+      <SliderComponent autoplay={true} initialItem={3} />
+    );
+
+    // validate initial active slide
+    expect(
+      document.querySelectorAll('li')[2].classList.contains('active')
+    ).toBeTruthy();
+    expect(asFragment()).toMatchSnapshot();
+
+    // validate autoplay
+    jest.advanceTimersByTime(6000);
+    expect(
+      document.querySelectorAll('li')[0].classList.contains('active')
+    ).toBeTruthy();
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it(`should not go to next slide after 6000ms when autoplay is falsy`, () => {
+    jest.useFakeTimers();
+    const { asFragment } = render(<SliderComponent autoplay={false} />);
+
+    // validate initial active slide
+    expect(
+      document.querySelectorAll('li')[0].classList.contains('active')
+    ).toBeTruthy();
+    expect(asFragment()).toMatchSnapshot();
+
+    // validate autoplay
+    jest.advanceTimersByTime(6000);
+    expect(
+      document.querySelectorAll('li')[0].classList.contains('active')
+    ).toBeTruthy();
+    expect(asFragment()).toMatchSnapshot();
+  });
+});
+
+describe('<Slider/> prop afterChange', () => {
+  test(`when active slide is the first slide`, () => {
+    const afterChange = jest.fn();
+    const { queryByTestId } = render(
+      <SliderComponent afterChange={afterChange} />
+    );
+
+    const leftArrow = queryByTestId('slider_left-arrow');
+    const rightArrow = queryByTestId('slider_right-arrow');
+    const firstDot = document.querySelectorAll('li')[0];
```

Comment:
How about the other dots?

Reply:
my idea is that this case is testing `afterChange`, so I only test the things can trigger it based on the implementation of the component. For an example based testing, I think this is enough.

Reply:
True